### PR TITLE
Cover Navigation Camera functionality with tests.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

PR adds test coverage for Navigation Camera related functionality.

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationCamera.swift [hunk 1]
```diff
@@ -125,10 +125,15 @@ public class NavigationCamera: NSObject, ViewportDataSourceDelegate {
      Call to this method executes a transition to `NavigationCameraState.following` state.
      When started, state will first change to `NavigationCameraState.transitionToFollowing` and then
      to the final `NavigationCameraState.following` when ended.
+     
+     - parameter completion: Completion handler, which is called whenever transition ends or doesn't
+     occur at all (e.g. in case if already in `NavigationCameraState.transitionToFollowing` or
+     `NavigationCameraState.following` state).
      */
-    public func follow() {
+    public func follow(_ completion: (() -> Void)? = nil) {
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationCamera.swift:133]: Decided to add this optional completion handler to public API to be able to test behavior more easily.

File: Tests/MapboxNavigationTests/NavigationCameraTests.swift [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+        let expectation = self.expectation(forNotification: .routeControllerProgressDidChange,
+                                           object: self) { _ in
+            return true
+        }
+        
+        NotificationCenter.default.post(name: .routeControllerProgressDidChange,
+                                        object: self,
+                                        userInfo: userInfo)
+        
+        wait(for: [expectation], timeout: 1.0)
```
Comment (S2Ler) [inline Tests/MapboxNavigationTests/NavigationCameraTests.swift:334]: For you information, you can wait for all expectations with `open func waitForExpectations(timeout: TimeInterval, handler: XCWaitCompletionHandler? = nil)`

**Review by S2Ler — APPROVED** [review]

File: Sources/MapboxNavigation/NavigationCamera.swift (hunk 1 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationCamera.swift:133]: There is no need to do this here as it's public API anyway.

File: Tests/MapboxNavigationTests/NavigationCameraTests.swift [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        followingMobileCamera.bearing = 0.0
+        followingMobileCamera.padding = .zero
+        followingMobileCamera.zoom = 15.0
+        followingMobileCamera.pitch = 45.0
+        
+        let cameraOptions = [
+            CameraOptions.followingMobileCamera: followingMobileCamera
+        ]
+        
+        delegate?.viewportDataSource(self, didUpdate: cameraOptions)
```
[the remark this answers is not in the document]
Comment (MaximAlien) [reply Tests/MapboxNavigationTests/NavigationCameraTests.swift]: Good catch. Forgot to update it.

**Pull request merged**